src/Raven.Server/Documents/TransactionOperationsMerger.cs

```diff
@@ -849,6 +850,7 @@ private enum PendingOperations
                                 rejectedBuffer.Add(operationToReject);
                             }
                             NotifyOnThreadPool(rejectedBuffer);
+                            GlobalFlushingBehavior.GlobalFlusher.Value?.MaybeFlushEnvironment(context.Environment); // commit stage 2 won't trigger so we need to ask for flush here
```

Comment:
I wonder if we shouldn't wait a bit after we schedule the flush / sync. Maybe same behavior as we have for indexes:

https://github.com/ravendb/ravendb/blob/3ec9c0e4cd21c945b48e0ba58c7b204233fa518e/src/Raven.Server/Documents/Indexes/Index.cs#L1252-L1258

?

Reply:
This flushes only the current environment while the high dirty memory check is for all environments